### PR TITLE
Styled and updated the sections components

### PR DESCRIPTION
**Pull request by natashapl**
### The changes in this pull request include the following:

- Updated the fieldset component for storybook
- Styled the Sections components

### CODE REVIEW

File: packages/design/sass/styles.scss [hunk 1]
```diff
@@ -166,3 +166,60 @@ iframe:focus {
   padding-bottom: 62.5%;
   height: 0;
 }
+
+.usa-fieldset {
```
Comment (danielnaab) [inline packages/design/sass/styles.scss:170]: This stylesheet will override the base USWDS classes that any end-user uses, which would be unexpected. Can we use utility classes for this in the fieldset component? (Or if absolutely necessary, scope the styles to a [BEM-style modifier](https://getbem.com/introduction/) like `usa-fieldset--a2j`?)

File: packages/design/sass/styles.scss [hunk 2]
```diff
@@ -166,3 +166,60 @@ iframe:focus {
...
+  .usa-legend {    
+    font-weight: normal;
+    line-height: 1.5rem;
+    text-transform: uppercase;
+  }
+
+  h3.section-heading-h3 {
+    font-size: 1.375rem;
+    padding-top: 0.5rem;
+    padding-bottom: 0.5rem;
```
Comment (danielnaab) [inline packages/design/sass/styles.scss:187]: On all the numeric values, can we use the USWDS token-derived utility classes, or if not possible, the SCSS mixins?

Ideally, there wouldn't be any magic numbers in the styles, and we inherit everything from the USWDS theme. To clarify, I'm referring to these: [Text utilities](https://designsystem.digital.gov/utilities/font-size-and-family/#utility-mixins)

File: packages/design/sass/styles.scss [hunk 3]
```diff
@@ -166,3 +166,60 @@ iframe:focus {
...
+  input,
+  select {
+    background: #f4f1f9;
+    font-weight: 700;
+  }
+
+  .usa-fieldset {
+    margin-top: 0;
+  }
+}
```
Comment (danielnaab) [inline packages/design/sass/styles.scss:209]: Same here, there are color theme tokens and corresponding utilities (eg, `bg-primary-light`), text utilities (`text-bold`), and margin (`margin-top-0`).

**Review by danielnaab — CHANGES_REQUESTED** [review]
Awesome, this looks really good! I'm hoping we can make more use of the utilities... where that's not possible, would you be okay with using something like `styled-components` to scope custom styles to the component, or use BEM naming conventions?

File: packages/design/sass/styles.scss (hunk 2 above)
Comment (natashapl) [reply packages/design/sass/styles.scss:187]: I wanted to do that but for some reason the token for numeric value and the ones for colors doesn't work for me.

File: packages/design/sass/styles.scss (hunk 3 above)
Comment (natashapl) [reply packages/design/sass/styles.scss:209]: The utility class for colors are not specific enough for certain backgrounds, such as the light violet that's in the focused input field when in edit mode. When I attempted to use color: color('violet-5'), it wasn't recognized. I was forced to report to the hex code.

**Review by danielnaab — APPROVED** [review]
Looks great!

**Pull request merged**